semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/RangeFinder.java

```diff
@@ -0,0 +1,68 @@
+package com.sourcegraph.semanticdb_javac;
+
+import com.sun.source.tree.CompilationUnitTree;
+import com.sun.source.tree.LineMap;
+import com.sun.source.util.TreePath;
+import com.sun.source.util.Trees;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.Name;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.sourcegraph.semanticdb_javac.Debugging.pprint;
+
+public class RangeFinder {
+  private static final HashMap<CompilationUnitTree, String> contentsCache = new HashMap<>();
```

Comment:
We should avoid static caches at all costs because they end up holding lots of memory. I don't think we even need a cache, we can store `SemanticdbVisitor.semanticdbText()` in a field.

Reply:
Yep, this shouldve been obvious :facepalm: Not sure I follow with `semanticdbText()` though :thinking:

Reply:
I think we can store the source code string in the constructor here https://github.com/sourcegraph/lsif-java/blob/48bce73664383381c85211a781a427924abae1a2/semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/SemanticdbVisitor.java#L48

Reply:
Ah yes I see that now :+1: